engine/access/rpc/backend/backend_stream_transactions_test.go

```diff
@@ -0,0 +1,364 @@
+		Communicator:             NewNodeCommunicator(false),
+		AccessMetrics:            metrics.NewNoopCollector(),
+		Log:                      s.log,
+		TxErrorMessagesCacheSize: 1000,
+		BlockTracker:             s.blockTracker,
+		SubscriptionParams: SubscriptionParams{
+			SendTimeout:    subscription.DefaultSendTimeout,
+			SendBufferSize: subscription.DefaultSendBufferSize,
+			ResponseLimit:  subscription.DefaultResponseLimit,
+			Broadcaster:    s.broadcaster,
+		},
+		TxResultsIndex: NewTransactionResultsIndex(s.transactionResults),
+	}
+}
+
+func (s *TransactionStatusSuite) addNewFinalizedBlock(parent *flow.Header, options ...func(*flow.Block)) {
+	s.finalizedBlock = unittest.BlockWithParentFixture(parent)
+	for _, option := range options {
+		option(s.finalizedBlock)
+	}
+
+	s.state.On("AtBlockID", mock.Anything).Unset()
+	s.state.On("AtBlockID", mock.Anything).Return(s.finalSnapshot, nil)
+
+	s.blocksArray = append(s.blocksArray, s.finalizedBlock)
+	s.blockMap[s.finalizedBlock.Header.Height] = s.finalizedBlock
+
+	s.broadcaster.Publish()
+}
+
+// TestSubscribeTransactionStatus tests the functionality of the SubscribeTransactionStatuses method in the Backend.
+// It covers the emulation of transaction stages from pending to sealed, and receiving status updates.
+func (s *TransactionStatusSuite) TestSubscribeTransactionStatus() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// Create subscription context with closer
+
+	// Generate sent transaction with ref block of the current finalized block
+	transaction := unittest.TransactionFixture()
+	transaction.SetReferenceBlockID(s.finalizedBlock.ID())
+	col := flow.CollectionFromTransactions([]*flow.Transaction{&transaction})
+	guarantee := col.Guarantee()
+	light := col.Light()
+	txId := transaction.ID()
+
+	expectedMsgIndexCounter := counters.NewMonotonousCounter(0)
+
+	// Create a special common function to read subscription messages from the channel and check converting it to transaction info
+	// and check results for correctness
+	checkNewSubscriptionMessage := func(sub subscription.Subscription, expectedTxStatus flow.TransactionStatus) {
+		unittest.RequireReturnsBefore(s.T(), func() {
+			v, ok := <-sub.Channel()
+			require.True(s.T(), ok,
+				"channel closed while waiting for transaction info:\n\t- txID %x\n\t- blockID: %x \n\t- err: %v",
+				txId, s.finalizedBlock.ID(), sub.Err())
+
+			txInfo, ok := v.(*convert.TransactionSubscribeInfo)
+			require.True(s.T(), ok, "unexpected response type: %T", v)
+
+			assert.Equal(s.T(), txId, txInfo.ID)
+			assert.Equal(s.T(), expectedTxStatus, txInfo.Status)
+
+			expectedMsgIndex := expectedMsgIndexCounter.Value()
+			assert.Equal(s.T(), expectedMsgIndex, txInfo.MessageIndex)
+			wasSet := expectedMsgIndexCounter.Set(expectedMsgIndex + 1)
+			require.True(s.T(), wasSet)
+		}, 50*time.Second, fmt.Sprintf("timed out waiting for transaction info:\n\t- txID: %x\n\t- blockID: %x", txId, s.finalizedBlock.ID()))
```

Comment:
50 seconds seems like a really long time. were you running into issues? I think a reasonable timeout within a unit test is 1-2 seconds